EssentialsGeoIP/src/com/earth2me/essentials/geoip/EssentialsGeoIPPlayerListener.java

```diff
@@ -205,4 +214,17 @@ private void downloadDatabase() {
             logger.log(Level.SEVERE, tl("connectionFailed"), ex);
         }
     }
-}
+
+    private boolean checkIfLocal(InetAddress address) {
+        if (address.isAnyLocalAddress() || address.isLoopbackAddress()) {
+            return true;
+        }
+
+        // Double checks if address is defined on any interface
+        try {
+            return NetworkInterface.getByInetAddress(address) != null;
```

Comment:
This throws `SocketException` only when an `I/O error occurs` (as per https://docs.oracle.com/javase/7/docs/api/java/net/NetworkInterface.html#getByInetAddress(java.net.InetAddress))

Should we really be catching this here, or should we be forwarding it up the call chain?

Reply:
These changes are not relevant to this PR. They were just an issue because I'm new to working with branches in git and sort of messed up.